Photos/app/src/main/java/hieunguyen/com/photos/FeedFragment.java

```diff
@@ -0,0 +1,128 @@
+
+        // Assign buttons so that button visibility setting in call to slice(list) line 48 will work
+        nextBtn = (Button) view.findViewById(R.id.feed_button_next);
+        nextBtn.setOnClickListener(new OnNextListener());
+
+        backBtn = (Button) view.findViewById(R.id.feed_button_back);
+        backBtn.setOnClickListener(new OnBackListener());
+
+        // If there are images to display, initiate the grid view
+        if (imageObjects.size() > 0) {
+            imageAdapter = new ImageAdapter(getActivity(), slice(imageObjects));
+
+            GridView image_grid = (GridView) view.findViewById(R.id.feed_image_grid);
+            image_grid.setAdapter(imageAdapter);
+            image_grid.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+                @Override
+                public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                    createDeleteDialog(position).show();
+                }
+            });
+        }
+
+        return view;
+    }
+
+    /**
+     * Creates dialog to confirm the user's decision to delete the image from feed/database
+     * @param position
+     * @return
+     */
+    public AlertDialog createDeleteDialog(final int position) {
+        AlertDialog.Builder deleteDialogBuilder = new AlertDialog.Builder(getActivity());
+        deleteDialogBuilder.setPositiveButton(R.string.confirm_dialog, new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dbService.delete(imageAdapter.getList().get(position));
+                imageObjects = dbService.getDataFromDB();
+                imageAdapter.setList(slice(imageObjects));
+            }
+        }).setNegativeButton(R.string.cancel, null).setMessage(R.string.delete_header);
+        return deleteDialogBuilder.create();
+    }
+
+    /**
+     * Utility function to slice the imageObject list to create paging and to adjust the view
+     * based on current page
+     * @param oldList The list of which to take a sublist
+     * @return The sliced list
+     */
+    public ArrayList<ImageDAO> slice(ArrayList<ImageDAO> oldList) {
+        // Decides whether to show 9 images or the rest of the list
+        int end = min((pageNumber + 1) * 9, oldList.size());
+
+        // Sets visibility of arrows depending on the page user is on
+        if (nextBtn != null) {
+            int visibility = (end == oldList.size() ? View.INVISIBLE : View.VISIBLE);
+            nextBtn.setVisibility(visibility);
+        }
+
+        if (backBtn != null) {
+            int visibility = (pageNumber == 0 ? View.INVISIBLE : View.VISIBLE);
+            backBtn.setVisibility(visibility);
+        }
+
+        return new ArrayList<>(oldList.subList(pageNumber*9, end));
+    }
+
+    /**
+     * Listener for feed_next_button
+     */
+    private class OnNextListener implements View.OnClickListener {
+        @Override
```

Comment:
This is an odd way of doing it, and i think you could make it a lot easier by not creating separate classes. Try out butterknife- it will save you a ton of time!